module/VuFind/src/VuFind/Db/Entity/AccessTokenInterface.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Entity model interface for access tokens.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+
+namespace VuFind\Db\Entity;
```

Comment:
Is there a reason this is not in the VuFind\Db\Interface namespace? That was established with the intent of holding interfaces like this one.

Reply:
(If you think the Interface namespace is pointless, I also wouldn't object to moving the existing interface into the Entity namespace).

Reply:
Would the `VuFind\Db\Interface` namespace hold interfaces for both entities and services? If so, then I guess it would be good to move the other three new interfaces in that namespace too? My personal preference would be to not have separate namespaces for interfaces. I think this is the first such namespace in VuFind?

Reply:
I hadn't really thought about service interfaces yet, though I think it's a good idea. Personally, at this point, I think I like `\VuFind\Db\Entity\UserEntityInterface` best. If you agree, I can rename `UserAccountInterface` and we can follow that pattern for new interfaces as we move this process forward.

Reply:
Great, thanks for talking this through. I'll open a PR to fix the naming as the next step after #3485 is merged.

Reply:
Re: personal preference, it definitely is not a strong one, especially with the naming convention you proposed, which does make things more clear.

Reply:
See #3500 -- as of this writing, I'm just waiting for my local test run to complete, but I'm pretty sure I didn't miss anything. :-)